cloudify_awssdk/ec2/resources/instances.py

```diff
@@ -0,0 +1,348 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+'''
+    EC2.Instances
+    ~~~~~~~~~~~~~~
+    AWS EC2 Instances interface
+'''
+
+# Common
+import json
+# Cloudify
+from cloudify import compute
+from cloudify import ctx
+from cloudify.exceptions import NonRecoverableError
+from cloudify_awssdk.common import decorators, utils
+from cloudify_awssdk.common.constants import EXTERNAL_RESOURCE_ID
+from cloudify_awssdk.ec2 import EC2Base
+# Boto
+from botocore.exceptions import ClientError, ParamValidationError
+
+RESOURCE_TYPE = 'EC2 Instances'
+RESERVATIONS = 'Reservations'
+INSTANCES = 'Instances'
+PENDING = 0
+RUNNING = 16
+SHUTTING_DOWN = 32
+TERMINATED = 48
+STOPPING = 64
+STOPPED = 80
+PS_OPEN = '<powershell>'
+PS_CLOSE = '</powershell>'
+GROUP_TYPE = 'cloudify.nodes.aws.ec2.SecurityGroup'
+NETWORK_INTERFACE_TYPE = 'cloudify.nodes.aws.ec2.Interface'
+SUBNET_TYPE = 'cloudify.nodes.aws.ec2.Subnet'
+GROUPIDS = 'SecurityGroupIds'
+NETWORK_INTERFACES = 'NetworkInterfaces'
+SUBNET_ID = 'SubnetId'
+INSTANCE_ID = 'InstanceId'
+INSTANCE_IDS = 'InstanceIds'
+
+
+class EC2Instances(EC2Base):
+    '''
+        EC2 Instances interface
+    '''
+    def __init__(self, ctx_node, resource_id=None, client=None, logger=None):
+        EC2Base.__init__(self, ctx_node, resource_id, client, logger)
+        self.type_name = RESOURCE_TYPE
+
+    @property
+    def properties(self):
+        '''Gets the properties of an external resource'''
+        params = {INSTANCE_IDS: [self.resource_id]}
+        try:
+            resources = \
+                self.client.describe_instances(**params)
+        except ClientError:
+            pass
+        else:
+            reservations = resources.get(RESERVATIONS, [])
+            instances = []
+            for r in reservations:
+                for i in r.get(INSTANCES, []):
+                    instances.append(i)
+            if len(instances) == 1:
+                return instances[0]
+        return None
+
+    @property
+    def status(self):
+        '''Gets the status of an external resource'''
+        props = self.properties
+        if not props:
+            return None
+        return props['State']['Code']
+
+    def create(self, params):
+        '''
+            Create AWS EC2 Instances.
+        '''
+        self.logger.debug('Creating %s with parameters: %s'
+                          % (self.type_name, params))
+        res = self.client.run_instances(**params)
+        self.logger.debug('Response: %s' % res)
```

Comment:
@EarthmanT I think we can use ``.format`` instead of  ``%`` because ``%`` is deprecated